### PR TITLE
Migrate string-refs

### PR DESCRIPTION
**Pull request by Raj-Datta-Manohar**
Removed usage of this.refs.customcheck and this.refs.custommentions  

Fixes https://github.com/mattermost/mattermost-server/issues/15844

### CODE REVIEW

File: components/user_settings/notifications/user_settings_notifications.jsx [hunk 1]
```diff
@@ -141,6 +141,8 @@ export default class NotificationsTab extends React.PureComponent {
         super(props);
 
         this.state = getNotificationsStateFromProps(props);
+        this.customcheck = React.createRef();
```
Comment (hahmadia) [inline components/user_settings/notifications/user_settings_notifications.jsx:144]:
```suggestion
        this.customCheck = React.createRef();
```

File: components/user_settings/notifications/user_settings_notifications.jsx [hunk 2]
```diff
@@ -141,6 +141,8 @@ export default class NotificationsTab extends React.PureComponent {
...
+        this.custommentions = React.createRef();
```
Comment (hahmadia) [inline components/user_settings/notifications/user_settings_notifications.jsx:145]:
```suggestion
        this.customMentions = React.createRef();
```

**Review by hahmadia — CHANGES_REQUESTED** [review]
Thanks for contribution! Code looks good for most part but we follow camelCase naming convention so that's all that needs changed. Thanks! Once change, let me know and I will approve the PR. 👍

**Review by hahmadia — APPROVED** [review]
LGTM!

File: components/user_settings/notifications/user_settings_notifications.jsx [hunk 3]
```diff
@@ -141,6 +141,8 @@ export default class NotificationsTab extends React.PureComponent {
         super(props);
 
         this.state = getNotificationsStateFromProps(props);
+        this.customCheck = React.createRef();
+        this.customMentions = React.createRef();
```
Comment (Willyfrog) [inline components/user_settings/notifications/user_settings_notifications.jsx:145]: in order to be consistent with the rest of PRs, could you rename the references?
```suggestion
        this.customCheckRef = React.createRef();
        this.customMentionsRef = React.createRef();
```

**Review by Willyfrog — CHANGES_REQUESTED** [review]
Hi! thanks for contributing!
I left a comment for consistency regarding this kind of PRs
Could you also remove the comment for disabling the slint rule in https://github.com/mattermost/mattermost-webapp/pull/6685/files#diff-26695f8c40a0d3beab0e24d9ab748a22R3

**Comment by Willyfrog** [comment]
fixes https://github.com/mattermost/mattermost-server/issues/15844

**Comment by Willyfrog** [comment]
> Hi! thanks for contributing!
> I left a comment for consistency regarding this kind of PRs
> Could you also remove the comment for disabling the slint rule in https://github.com/mattermost/mattermost-webapp/pull/6685/files#diff-26695f8c40a0d3beab0e24d9ab748a22R3

could you do this too? so it prevents people from using string-refs in the future

**Review by Willyfrog — CHANGES_REQUESTED** [review]
seems we missed a couple of refs and for that the tests are failing.
see
https://github.com/mattermost/mattermost-webapp/blob/d69380ad78872b64487a4dc429cf7c10bca53acd/components/user_settings/notifications/user_settings_notifications.jsx#L891
https://github.com/mattermost/mattermost-webapp/blob/d69380ad78872b64487a4dc429cf7c10bca53acd/components/user_settings/notifications/user_settings_notifications.jsx#L914

**Comment by Raj-Datta-Manohar** [comment]
Thanks @Willyfrog !

**Comment by hmhealey** [comment]
Removing myself for reviews since we've got two dev reviewers already.

**Review by Willyfrog — COMMENTED** [review]
LGTM! thanks for contributing :)

**Review by Willyfrog — APPROVED** [review]

**Comment by jgilliam17** [comment]
Thank you @Raj-Datta-Manohar 
I am not seeing `Automatic Direct Message Replies` section in Account Settings>Notifications 
Can you please take a look? Thanks.
Screen Shot 2020-10-20 at 10 29 40 AM

**Comment by jgilliam17** [comment]
@Raj-Datta-Manohar - Ah, I didn't check the System Console setting for Auto replies, it was off 😞
Sorry about that!

Tested, looks good to merge.
- Verified notifications, words that trigger mentions, custom notifications etc. - as expected.
@Willyfrog can you please help merge? Thanks.

**Comment by Raj-Datta-Manohar** [comment]
@jgilliam17 , wasn't online, glad that it's figured out

**Pull request merged**